pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
Sorry my intention wasn't clear. This is something I would like to see fixed, but in a separate PR because this PR isn't introducing this pattern. Sorry if this held this up. I'm fine with this PR given that this exists already.